mixer/adapter/kubernetesenv/kubernetesenv.go

```diff
@@ -299,3 +351,52 @@ func newKubernetesClient(kubeconfigPath string, env adapter.Env) (k8s.Interface,
 	}
 	return k8s.NewForConfig(config)
 }
+
+func (b *builder) createCacheController(k8sInterface k8s.Interface, clusterID string) error {
+	controller, err := getNewCacheController(b, k8sInterface, b.kubeHandler.env)
+	if err == nil {
+		b.controllers[clusterID] = controller
+		b.kubeHandler.k8sCache[clusterID] = controller
+		b.kubeHandler.env.Logger().Infof("created remote controller %s", clusterID)
+	} else {
+		b.kubeHandler.env.Logger().Errorf("error on creating remote controller %s err = %v", clusterID, err)
+	}
+
+	return err
+}
+
+func (b *builder) deleteCacheController(clusterID string) error {
+	b.kubeHandler.k8sCache[clusterID].StopControlChannel()
+	delete(b.controllers, clusterID)
+	delete(b.kubeHandler.k8sCache, clusterID)
+
+	b.kubeHandler.env.Logger().Infof("deleted remote controller %s", clusterID)
+
+	return nil
+}
+
+func initMultiClusterSecretController(b *builder, kubeconfig string, env adapter.Env) (err error) {
+	var clusterNS string
+
+	paramsProto := b.adapterConfig
+	if paramsProto.ClusterRegistriesNamespace == "" {
```

Comment:
super nit: I believe this could be written as:

```
if clusterNs = paramsProto.ClusterRegistriesNamespace; clusterNs == "" {
    if clusterNs = os.Getenv("POD_NAMESPACE"); clusterNs == "" {
        clusterNs = defaultClusterRegistriesNamespace
    }
}
```

Would save the `else` clause.

Reply:
done. thanks, much cleaner